BMC - implement OAuth2 and add bmc-itsm-add-attachment command (#43166)
demisto/content: Packs/BmcITSM/Integrations/BmcITSM/README.md

| Incident type |  | False |
| Fetch incidents |  | False |

## Prerequisites: Register the Client

### 1. Registering Non-Native (Standard) Clients

To begin, log in to the RSSO Administration Console and navigate to the OAuth2 section.  

> Before registering a specific client, ensure your general settings, such as access token timeout, refresh token timeout, and the RSSO server URL are correctly configured.  

Reviewer: ```suggestion
2. Ensure general settings such as access token timeout, refresh token timeout, and the RSSO server URL are correctly configured.  
```

For the official BMC registration demo, watch the [video](https://www.youtube.com/watch?v=9DVAvO-Zs_w&list=PLLwZptFDIb_texh1fnPprXOB5-pq3ZiO5&index=13) between 28:30 and 31:10

**Navigation:** OAuth2 > Clients > Register Client

---

> #### Fields

Reviewer: ```suggestion
```

| Field | Description |

> |------|------------|

Reviewer: ```suggestion
  |------|------------|
```

| **Client Name & Enabled** | Name the client and check **Enabled** to allow authentication. |
| **Token Exchange** | (Optional) Allows the client to request security tokens. |
| **Redirect URIs** | URIs for the auth code. Max 2000 chars. For BMC SSO agents, use: `http(s)://<domain>:[port]/[path]/_rsso/oauth/callback` |
| **Trusted Clients** | (Optional) List IDs eligible to validate this client’s tokens. |

> | **OpenID** | (Optional) Required for OIDC or authenticating across different domains. |

Reviewer: ```suggestion
  | **OpenID** | (Optional) Required for OIDC or authenticating across different domains. |
```

| **Token Timeouts** | (Optional) Define custom access/refresh token limits (overrides tenant settings). |
| **Multi-Tenant** | (SaaS only) Select checkbox, then add specific Redirect URIs per tenant. |

> **CAUTION**

Reviewer: ```suggestion
  **Note:**  
```

> **Client Secret**: Generated only after clicking Save. Copy it immediately, it will not be displayed again. It is used to sign the JWT.

Reviewer: ```suggestion
  The **Client Secret** is generated only after clicking **Save**; copy it immediately as it cannot be retrieved later and is required to sign the JSON Web Token (JWT).
```

---

### 2. Managing Client Secrets (Non-Native Only)

If you lose a secret or need a specific length (36–256 ASCII characters), you can manually generate up to **3 keys**:

1. Edit the client > **Additional secrets** > **Add secret**.

> 2. Enter the new key and **copy it before** clicking Confirm (it will be obfuscated after).

Reviewer: ```suggestion
2. Enter the new key and copy it before clicking **Confirm** (it will be obfuscated after).
```
